scripts/ubi8-sysroot.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+TARGET_ARCH="$1"
+UBI_VERSION="$2"
+
+SYSROOT_PATH="/sysroot/ubi-$UBI_VERSION"
+
+ubi_packages="systemd-libs systemd-devel glibc glibc-devel gcc libstdc++-devel libstdc++-static kernel-headers"
+
+apt-get update 1>&2 && apt-get install --no-install-recommends -y dnf 1>&2
+dnf install --releasever=8 --forcearch="${TARGET_ARCH}" \
```

Comment:
This is interesting. Is this using a Debian package manager to down load a Red Hat package manager?

Reply:
Yup, dnf supports alternative install roots, so it's a bit like debootstrap or pacstrap from debian and arch respectively in that it will allow you to install packages into a filesystem subtree.

The general approach means we can:
* Grab the actual packaged libraries and headers from pretty much any linux distro and for any ISA/architecture (or even windows)
* Install them under a directory (commonly referred to as a sysroot)
* Configure the C, C++ and Rust compilers to look for the headers/libraries under the sysroot rather than the actual root directory

Once have that (and cross compilers/binutils on the host, ie rustc, clang, lld, llvm-strip etc) you can build for all of your targets from the same base toolchain.

It's very commonly used when doing any sort of embedded work or compiling for systems that you don't have high performance hardware available for